tests/CalculationsTest.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace MichaelRubel\Couponables\Tests;
+
+use MichaelRubel\Couponables\Exceptions\InvalidCouponTypeException;
+use MichaelRubel\Couponables\Models\Coupon;
+
+class CalculationsTest extends TestCase
+{
+    /** @test */
+    public function testCanCalcUsingSubtractionStrategy()
+    {
+        $coupon = Coupon::create([
+            'code'  => 'test-code',
+            'type'  => 'subtraction',
+            'value' => '250', // <-- Amount to subtract.
+        ]);
+
+        $newPrice = $coupon->calc(value: 500);
+        // 500.00 - 250.00 = 250.00
+
+        $this->assertSame(250.00, $newPrice);
+    }
+
+    /** @test */
+    public function testCanCalcUsingPercentStrategy()
+    {
+        $coupon = Coupon::create([
+            'code'  => 'test-code',
+            'type'  => 'percentage',
+            'value' => '50', // <-- %50.
```

Comment:
Suggest to change test value to different than 50%.
Case of (discount/100)*value & (1-discount/100)*value will pass at this case